source/class.cpp

```diff
@@ -868,10 +868,15 @@ string bind_forward_declaration(CXXRecordDecl const *C, Context &context)
 
 	string const include = relevant_include(C);
 
+	string holder_type = Config::get().holder_type();
+
 	string c = "\t// Forward declaration for: " + qualified_name + " file:" + (include.size() ? include.substr(1, include.size() - 2) : "") + " line:" + line_number(C) + "\n";
 
-	string maybe_holder_type = ", std::shared_ptr<{}>"_format(qualified_name);
-	if( is_inherited_from_enable_shared_from_this(C) ) maybe_holder_type = ", std::shared_ptr<{}>"_format(qualified_name);
+	string maybe_holder_type = ", {}<{}>"_format(holder_type, qualified_name);
```

Comment:
Optional: looks like original code have duplication, this could be refactored as `string maybe_holder_type;`

Reply:
Are we sure that it will be always correct to replace:
`string maybe_holder_type = ", {}<{}>"_format(holder_type, qualified_name);`
with
`string maybe_holder_type;`
?

The following logic that sets `maybe_holder_type` is 
```
if {
}
else if{
  if
}
```

I think that I would be in favor of leaving the logic as in the original code and merge it as it is.
Would you be fine with this?

Reply:
Yes, let's leave the old logic for now and i will refactor this a bit later.